src/go/rdctl/pkg/config/config.go

```diff
@@ -76,56 +69,52 @@ func DefineGlobalFlags(rootCmd *cobra.Command) {
 	rootCmd.PersistentFlags().StringVar(&configPath, "config-path", "", fmt.Sprintf("config file (default %s)", DefaultConfigPath))
 	rootCmd.PersistentFlags().StringVar(&connectionSettings.User, "user", "", "overrides the user setting in the config file")
 	rootCmd.PersistentFlags().StringVar(&connectionSettings.Host, "host", "", "default is 127.0.0.1; most useful for WSL")
-	rootCmd.PersistentFlags().StringVar(&connectionSettings.Port, "port", "", "overrides the port setting in the config file")
+	rootCmd.PersistentFlags().IntVar(&connectionSettings.Port, "port", 0, "overrides the port setting in the config file")
 	rootCmd.PersistentFlags().StringVar(&connectionSettings.Password, "password", "", "overrides the password setting in the config file")
 }
 
-// GetConnectionInfo returns the connection info if it has it, and an error message explaining why
-// it isn't available if it doesn't have it.
-// So if the user runs an `rdctl` command after a factory reset, there is no config file (in the default location),
-// but it might not be necessary. So only use the error message for the missing file if it is actually needed.
+// GetConnectionInfo returns the connection details of the application API server.
```

Comment:
I think we should say that the function returns `os.ErrNotExist` when it cannot find the config file and the user has not specified a non-default config file. This is important information to callers of this function.

Reply:
I added a `mayBeMissing` parameter. If `true`, the function will return `nil, nil` when the default config file is missing. If `false`, it will return the `os.ErrNotExist` error for the default file.

Note that I also modified the `shutdown` command to use `mayBeMissing = true`. All other non-snapshot callers use `mayBeMissing = false`.